services/RetourService.php

```diff
@@ -108,14 +113,56 @@ public function findRedirectMatch($url)
         // Look up the static redirects next
         $redirects = null;
         $redirects = $this->getAllStaticRedirects();
-        $result = $this->lookupRedirect($url, $redirects);
+        $result = $this->lookupRedirect($url, $redirects, true);
+
         if ($result) {
+            if ($result['locale'] == 'all' && !$this->checkCachedRedirectsForEachLocale($url)) {
+                //There's a chance that it found an "all" but
+                //there's also at least one specific match still lurking...
+                //cache those as well, and if it turns out it's actually the
+                //redirect we wanted, use it instead.
+                $query = craft()->db->createCommand()
+                    ->select('*')
+                    ->from('retour_static_redirects')
+                    ->where('redirectSrcUrl = "' . $url . '"')
```

Comment:
I have concerns about doing this here, I feel we're opening ourselves up to injection

Reply:
Oh yeah great call!

I was so happy this _seemed_ to be working, I didn't think about that aspect.

I can sanitize that pretty easily (I think 🤔 ) and I'll re-up when I get it 🙇

Reply:
Easy! https://www.yiiframework.com/doc/guide/2.0/en/db-dao#binding-parameters

♻️

Reply:
👏 thanks!